R/QFeatures-filter.R

```diff
@@ -309,6 +309,8 @@ filterFeaturesWithFormula <- function(object, filter, na.rm, keep, ...) {
 
 ## Internal function that 
 .checkFilterVariables <- function(object, vars) {
+    ## Ignore variables from the user environment
+    vars <- vars[!vars %in% ls(envir = parent.frame(4))]
```

Comment:
Yes it is always four because we only want to include variables from the environment the function was called in (may not always be `.GlobalEnv`). Here is a "traceback" counter:

0 in `.checkFilterVariables()`
1 in `FilterFeaturesWithFormula()`
2 in `.local()`
3 in `filterFeatures()`
4 in environment the function was called

Reply:
I'm improving the unit test to ensure this is checked.

Reply:
Could you also add this information as a comment in the code.